### PR TITLE
INVESTIGATE PAGINATION ADJUSTMENTS: As Dan, I want the extraction method updated, so that I can work with API that have varying requirements.

### PR DESCRIPTION
**Pull request by richardmatthewsdev**
**Acceptance Criteria**
- a flexible solution for building extraction definitions is identified
- follow up stories are drafted

### CODE REVIEW

File: app/supplejack/extraction/execution.rb [hunk 1]
```diff
@@ -40,6 +35,14 @@ def call
 
     private
 
+    def extract_and_save_document(request)
+      @de = DocumentExtraction.new(request, @extraction_job.extraction_folder, @previous_request&.body)
+      @previous_request = @de.extract
+      @de.save
```
Comment (motizuki) [inline app/supplejack/extraction/execution.rb:41]: do we need to check if this was successful?

File: app/controllers/parameters_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class ParametersController < ApplicationController
+  before_action :find_parameter, only: %i[update destroy]
+
+  def create
+    @parameter = Parameter.new(parameter_params)
+
+    @parameter.save!
```
Comment (motizuki) [inline app/controllers/parameters_controller.rb:9]: is it ok to `!` here?

File: app/controllers/parameters_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    render json: @parameter
+  end
+
+  def update
+    @parameter.update(parameter_params)
```
Comment (motizuki) [inline app/controllers/parameters_controller.rb:15]: do we need to check for true/false?

File: app/controllers/parameters_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    render json: @parameter
+  end
+
+  def destroy
+    @parameter.destroy!
```
Comment (motizuki) [inline app/controllers/parameters_controller.rb:21]: save as abovce

**Review by motizuki — APPROVED** [review]
Chunky

File: app/controllers/parameters_controller.rb (hunk 3 above)
Comment (richardmatthewsdev) [reply app/controllers/parameters_controller.rb:15]: We essentially take whatever they have given us so we don't really need to check :).

File: app/supplejack/extraction/execution.rb (hunk 1 above)
Comment (richardmatthewsdev) [reply app/supplejack/extraction/execution.rb:41]: Na it's up to the harvester to decide if something has gone wrong with the extraction and fix the issue themselves.

**Comment by richardmatthewsdev** [comment]
> Chunky

Yeah sorry this got much larger than expected!

File: app/frontend/js/ExtractionDefinitionFormLogic.js [hunk 5]
```diff
@@ -1,49 +1,63 @@
...
-  const tokenisedElements = document.getElementsByClassName('js-extraction-definition-tokenised-form');
-
-  extractionDefinitionPaginationLogic(extractionDefinitionPaginationTypeSelect.value);
-
-  extractionDefinitionPaginationTypeSelect.addEventListener('change', (event) => {
-    extractionDefinitionPaginationLogic(event.target.value);
-  });
+import { each } from "lodash";
+
+const extractionDefinitionPaginationTypeSelect = document.getElementById(
```
Comment (richardmatthewsdev) [inline app/frontend/js/ExtractionDefinitionFormLogic.js:3]: Remove more of me!

File: app/frontend/js/apps/ExtractionApp/components/HeaderActions.jsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    "col-12": !appDetails.extractionDefinition.paginated,
+  });
+
+  return createPortal(
+    <>
+      <button className="btn btn-success" onClick={handlePreviewClick}>
+        <i className="bi bi-play" aria-hidden="true"></i> Preview
+      </button>
+
+      <Modal
```
Comment (richardmatthewsdev) [inline app/frontend/js/apps/ExtractionApp/components/HeaderActions.jsx:69]: Display the Modal into another portal, probably at the bottom of the page :D

File: app/frontend/js/apps/ExtractionApp/components/Parameter.jsx [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+                    onChange={(e) => setContentValue(e.target.value)}
+                  />
+                )}
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <Modal show={showModal} onHide={handleClose}>
```
Comment (richardmatthewsdev) [inline app/frontend/js/apps/ExtractionApp/components/Parameter.jsx:264]: Put Modal into the same Modal div :D

File: app/helpers/parameters_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module ParametersHelper
```
Comment (richardmatthewsdev) [inline app/helpers/parameters_helper.rb:1]: Delete me!

File: app/helpers/requests_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module RequestsHelper
```
Comment (richardmatthewsdev) [inline app/helpers/requests_helper.rb:1]: Delete me!

File: spec/factories/extraction_definition.rb [hunk 10]
```diff
@@ -3,16 +3,13 @@
...
-    pagination_type { 'page' }
-    page_parameter { 'page' }
-    page { 1 }
-    per_page_parameter { 'per_page' }
     per_page { 50 }
     total_selector { '$.totalObjects' }
+    page { 1 }
+    paginated { false }
 
     trait :ngataonga do
```
Comment (richardmatthewsdev) [inline spec/factories/extraction_definition.rb:14]: remove all ngataonga from app

File: spec/helpers/parameters_helper_spec.rb [hunk 11]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (richardmatthewsdev) [inline spec/helpers/parameters_helper_spec.rb:1]: Remove

File: spec/helpers/requests_helper_spec.rb [hunk 12]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (richardmatthewsdev) [inline spec/helpers/requests_helper_spec.rb:1]: Remove

**Review by paul-mesnilgrente — APPROVED** [review]

**Pull request merged**